src/unit/test_networking.c

```diff
@@ -4,6 +4,305 @@
 
 #include <stdatomic.h>
 
+/* Mock structures and functions */
+typedef struct mockConnection {
+    connection conn;
+    int error;
+    char *buffer;
+    size_t buf_size;
+    size_t written;
+} mockConnection;
+
+/* Mock connWrite function */
```

Comment:
Sort of nitpicky, but this is a fake and not a mock, since it is an alternative simplified implementation. A mock would have configurable return objects.